hashintel/hash
# Enable margin-clicking to focus on nearby blocks

## 🌟 What is the purpose of this PR?

This PR enables margin-clicking on Pages to improve page editing UX. So users can easily click on a blank space on the page to focus on closest text, instead of directly clicking to the text or empty paragraphs.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1202796290628216/f) _(internal)_

## 🔍 What does this change?

- Update page layout by making ProseMirror editor full width, and giving it `padding-horizontal` to center the blocks inside the editor.
  - So we don't have to deal with margin-clicking logic. `ProseMirror` handles it itself. Only thing I needed to do to achieve this was making the `ProseMirror` full width, and give it some `padding-bottom` (`padding-bottom` is also from design)
- Update page content width according to design
- Move page icon above the title if screen smaller than tablet width to match the design

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Open a new page
2. Try margin-clicking to confirm that editor focuses on closest text

## 📹 Demo

> ⚠️  Subtle pink background color is to demonstrate the margin-clickable area for reviewing purposes. ⚠️

Before             |  After
:-------------------------:|:-------------------------:
![](https://user-images.githubusercontent.com/39553853/192497124-855bb71e-d10c-40c8-b32f-c12b114b4120.png)  |  ![](https://user-images.githubusercontent.com/39553853/192497074-c8c79ed3-f3f5-4235-9c7e-e843d073b1a4.png)
![](https://user-images.githubusercontent.com/39553853/192497137-eeab5046-1519-44ff-904c-cdc41c720026.png)  |  ![](https://user-images.githubusercontent.com/39553853/192497090-769a7966-1157-4665-a684-f67850a9deb4.png)
![](https://user-images.githubusercontent.com/39553853/192497197-17d3be29-a3c8-4868-a76d-95950ea57bd5.png)  |  ![](https://user-images.githubusercontent.com/39553853/192497170-5b511ffc-e973-413d-ae63-bd451211effd.png)
